platforms/windows/SQLitePlugin/SQLitePlugin.cpp

```diff
@@ -0,0 +1,511 @@
+    {
+        SerialReactDispatcher.Post(
+            [onFailure{ std::move(onFailure) }]()
+        {
+            onFailure("attach isn't supported on this platform");
+        });
+    }
+
+    void SQLitePlugin::Close(
+        CloseOptions options,
+        std::function<void(std::string)> onSuccess,
+        std::function<void(std::string)> onFailure) noexcept
+    {
+        SerialReactDispatcher.Post(
+            [
+                options{ std::move(options) },
+                onSuccess{ std::move(onSuccess) },
+                onFailure{ std::move(onFailure) },
+                weak_this = std::weak_ptr(shared_from_this())
+            ]()
+        {
+
+            if (auto strongThis = weak_this.lock()) {
+
+                if (options.Path == nullptr || options.Path.empty())
+                {
+                    onFailure("You must specify database path");
+                    return;
+                }
+
+                hstring absoluteDbPath{ ResolveDbFilePath(to_hstring(options.Path)) };
+
+                if (strongThis->OpenDBs.find(absoluteDbPath) == strongThis->OpenDBs.end())
+                {
+                    onFailure("Specified db was not open");
+                    return;
+                }
+
+                sqlite3* dbHandle = strongThis->OpenDBs[absoluteDbPath];
+
+                if (sqlite3_close(dbHandle) != SQLITE_OK)
+                {
+                    // C# implementation returns success if the DB failed to close
+                    // Matching the existing implementation
+                    std::string debugMessage = "SQLitePluginModule: Error closing database: " + to_string(absoluteDbPath) + "\n";
+                    OutputDebugStringA(debugMessage.c_str());
+                }
+
+                strongThis->OpenDBs.erase(absoluteDbPath);
+
+                onSuccess("DB Closed");
+            }
+        });
+
+    }
+
+    SQLitePlugin::SQLitePlugin() {
+
+    };
+
+    SQLitePlugin::~SQLitePlugin() {
+        for (auto& db : OpenDBs)
+        {
+            sqlite3_close(db.second);
+        }
+    };
+
+    void SQLitePlugin::DeleteDB(
+        DeleteOptions options,
+        std::function<void(std::string)> onSuccess,
+        std::function<void(std::string)> onFailure) noexcept
+    {
+        SerialReactDispatcher.Post(
+            [
+                options{ std::move(options) },
+                onSuccess{ std::move(onSuccess) },
+                onFailure{ std::move(onFailure) },
+                weak_this = std::weak_ptr(shared_from_this())
+            ]()
+        {
+            if (auto strongThis = weak_this.lock())
+            {
+                if (options.Path == nullptr || options.Path.empty())
+                {
+                    onFailure("You must specify database name");
+                    return;
+                }
+
+                hstring absoluteDbPath{ ResolveDbFilePath(to_hstring(options.Path)) };
```

Comment:
This code seems a little similar to the ::Close() method - is there a way to extract the common parts?

Reply:
Done